tests/parser/parser.cpp

```diff
@@ -1647,146 +1649,121 @@ TEST(Parser, parseResets)
-    EXPECT_EQ_ERRORS(expectedErrors, parser);
-}
+        // order not specified
+        "    <reset variable=\"variable2\" test_variable=\"variable4\">\n"
+        "      lost text here\n"
+        "      <test_value>\n"
+        "        <some_invalid_tag/>\n" // test_value should only contain mathml
+        "      </test_value>\n"
+        "      <reset_value>\n"
+        "        <some_other_invalid_tag/>\n" // reset_value should only contain mathml
+        "      </reset_value>\n"
+        "    </reset>\n"
 
-TEST(Parser, parseResetsCheckResetObjectCheckWhenObject)
-{
-    const std::string input =
-        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
-        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" id=\"mid\">\n"
-        "  <component name=\"component2\" id=\"c2id\">\n"
-        "    <variable name=\"variable1\" id=\"vid\"/>\n"
-        "    <variable name=\"V_k\" id=\"vid\"/>\n"
-        "    <reset variable=\"V_k\" order=\"a\" id=\"rid\">\n"
-        "      <when order=\"5.9\" goods=\"socks\">\n"
-        "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
-        "          some condition in mathml\n"
-        "        </math>\n"
-        "      </when>\n"
+        //order not specified, unknown attribute
+        "    <reset variable=\"variable2\" test_variable=\"variable4\" i_dont_belong_here=\"yep_really_i_dont\">\n"
+
+        //test_value missing
+        //reset_value missing
         "    </reset>\n"
         "  </component>\n"
         "</model>\n";
+
     const std::vector<std::string> expectedErrors = {
-        "Reset in component 'component2' referencing variable 'V_k' has a non-integer order value 'a'.",
-        "When in reset referencing variable 'V_k' with order '' has an invalid attribute 'goods'.",
-        "When in reset referencing variable 'V_k' with order '' does not have an order defined.",
-        "When in reset referencing variable 'V_k' with order '' contains only one MathML child element.",
-    };
+        "Reset referencing variable 'variable3' is not a valid reference for a variable in component 'component2'.",
+        "Reset referencing test_variable 'variable1' is not a valid reference for a variable in component 'component2'.",
+        "Reset in component 'component2' referencing variable '' and test_variable '' has an unexpected attribute in the test_value block of 'one_invalid_attribute'.",
+        "Reset in component 'component2' referencing variable '' and test_variable '' has an unexpected attribute in the reset_value block of 'another_invalid_attribute'.",
+        "Reset in component 'component2' referencing variable '' and test_variable '' has multiple test_value blocks.",
+        "Reset in component 'component2' referencing variable '' and test_variable '' has multiple reset_value blocks.",
+        "Reset has an invalid non-whitespace child text element '\n      lost text here\n      '.  Expected either 'test_value' or 'reset_value' block.",
```

Comment:
Inconsistency: use single quotes or not for `test_value` and `reset_value` blocks.

Otherwise, we might want to rephase the second part to read "Either a test_value or a reset_value block is expected." (Also, no double space after a period. We don't use mechanical typewriters anymore...)

Reply:
Mechanical typewriters are the best, long live the double space after fullstop!